### PR TITLE
Add upgrade guide for 2.6.0

### PR DESCRIPTION
**Pull request by nathandyer**
## Status

Ready for review

This PR is ready for review, but should not be merged until SecureDrop 2.6.0 is released. Because of this, I'm going to leave this as a "draft."

## Description of Changes

This change adds an upgrade guide for the 2.5.2 to 2.6.0 upgrade, and updates the `git checkout` commands in the install guide to refer to the latest release.

This also includes a two-line change to the `update_version.py` script to match some changes to the documentation structure that landed recently.

## Testing
* [x] CI passes
* [ ] Visual inspection

## Release 

This should be merged to coincide with the SecureDrop 2.6.0 release.

## Checklist (Optional)

- [x] Doc linting (`make docs-lint`) passed locally
- [ ] Doc link linting (`make docs-linkcheck`) passed
- [x] You have previewed (`make docs`) docs at http://localhost:8000

### CODE REVIEW

File: docs/upgrade/2.5.2_to_2.6.0.rst [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+Upgrade from 2.5.2 to 2.6.0
+===========================
+
+Update Servers to SecureDrop 2.6.0
+----------------------------------
+Servers running Ubuntu 20.04 will be updated to the latest version of SecureDrop
+automatically within 24 hours of the release.
+
+Update Workstations to SecureDrop 2.6.0
```
Comment (legoktm) [inline docs/upgrade/2.5.2_to_2.6.0.rst:11]: Should we add a reminder/pointer about the Tails advisory?

**Review by legoktm — CHANGES_REQUESTED** [review]
The version number is now also in `pyproject.toml`, that should be updated (and added to the version bump script)

Also you can delete the 2.3.x upgrade guides now I think

File: docs/upgrade/2.5.2_to_2.6.0.rst (hunk 1 above)
Comment (nathandyer) [reply docs/upgrade/2.5.2_to_2.6.0.rst:11]: I think it's a good idea! Just pushed a new commit with that added.

**Comment by nathandyer** [comment]
Thanks @legoktm! Those changes should now be in and ready for another review whenever you have time. Thank you!

File: docs/upgrade/2.5.2_to_2.6.0.rst [hunk 2]
```diff
@@ -11,14 +11,30 @@ automatically within 24 hours of the release.
...
+Before upgrading your Workstations to SecureDrop 2.6.0, we 
+strongly recommend that you first upgrade to Tails 5.14, which includes
+important updates to disk encryption and passphrase hashing algorithms. 
+
+We also recommend updating all other encrypted drives to LUKS2, and ensuring
+you have strong passphrases.
+
+We have issued a Security Advisory, which provides detailed instructions for
+updating the Workstations, as well as any other encrypted drives. You can find
+that `advisory on the SecureDrop blog.
```
Comment (legoktm) [inline docs/upgrade/2.5.2_to_2.6.0.rst:25]:
```suggestion
that `advisory on the SecureDrop website.
``` 

minor nit, but we don't call it the "blog" on the website, it's just "news".

**Review by legoktm — APPROVED** [review]
LGTM!

**Pull request merged**